tests/e2e/tests/pilot/util/environment.go

```diff
@@ -251,13 +310,28 @@ func (e *Environment) Setup() error {
 		return err
 	}
 
+	if e.Config.ClusterRegistriesDir != "" {
+		if err = e.createMulticlusterConfig(); err != nil {
+			return err
+		}
+	}
+
 	if e.Config.UseAutomaticInjection {
+		// Automatic side car injection is not supported when multiple clusters are being tested
+		// as all istio control plane components are only installed on the first cluster.
+		if e.RemoteKubeConfig != "" {
+			return err
```

Comment:
err is nil here... change to something like return `fmt.Errorf("Automatic side car injection is not supported when multiple clusters are being tested")`

Reply:
I added a log.Info here,  but since the rest of the file convention was to just return err I didn't follow your exact suggestion

Reply:
if err == nil then the tests will still try to run. IMO, you should set err = fmt.Errorf(...) here

Reply:
Line 307 should be something like:

  return fmt.Errorf("Automatic side car injection is not supported when multiple clusters are being tested")